File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SumOfSquares is the sum of the squares in range [1,n]
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// SquareOfSum is the sum of numbers in range [1,n], squared
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// Difference is the difference betwee sum of squares and square of sums
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int n)
{
    return n * (n + 1) * (2 * n + 1) / 6;
}

unsigned int square_of_sum(unsigned int n)
{
    n = (n * (n + 1)) / 2;
    return n * n;
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}
